build/gen_stub.php

```diff
@@ -1807,11 +1807,14 @@ private function getExampleSection(DOMDocument $doc): DOMElement {
         $refSec = $this->generateRefSect1($doc, 'examples');
 
         $example = $doc->createElement('example');
-        $example->setAttribute('xml:id', 'func-or-method-name.example.basic');
+        $fnName = $this->name->__toString();
+        $fnNameForId = strtolower($fnName);
+        $fnNameForId = str_replace(['_', '\\', '::'], ['-', '-', '.'], $fnNameForId);
+        $example->setAttribute('xml:id', $fnNameForId . '.example.basic');
```

Comment:
The ID of the example section should use the same code as the ID of the `refentry` (except for the `.example.basic` postfix)